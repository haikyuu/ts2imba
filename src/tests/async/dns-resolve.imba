import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
import * as dns from 'dns'

async function resolveDomain(domain) {
	const records = (await dns.resolveTxt(domain)).flat();
	return records;
}
'''

test 'dns resolve with flat' do
	const result = await build tsx-code
	expect(result.code).toContain('(await dns.resolveTxt(domain)).flat()')
	expect(result.code).not.toContain('await dns.resolveTxt(domain).flat()')
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()
