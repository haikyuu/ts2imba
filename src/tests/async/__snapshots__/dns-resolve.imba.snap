// Vitest Snapshot v1

exports[`dns resolve with flat 1`] = `
"import {* as dns} from \\"dns\\"
def resolveDomain(domain)
	const records = (await dns.resolveTxt(domain)).flat()
	records
"
`;
